chromium_src/third_party/blink/renderer/modules/credentialmanager/credentials_container.cc

```diff
@@ -0,0 +1,55 @@
+#include "third_party/blink/renderer/modules/credentialmanager/credentials_container.h"
+
+#include "gin/converter.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise_resolver.h"
+#include "third_party/blink/renderer/platform/bindings/v8_per_isolate_data.h"
+#include "v8/include/v8.h"
+
+namespace blink {
+
+  CredentialsContainer::CredentialsContainer() = default;
+
+  ScriptPromise CredentialsContainer::get(
+    ScriptState* script_state,
+    const CredentialRequestOptions& options) {
+      ScriptPromise::InternalResolver resolver(script_state);
+      ScriptPromise promise = resolver.Promise();
+      v8::Isolate* isolate = V8PerIsolateData::MainThreadIsolate();
```

Comment:
you can't assume the main thread isolate here. You should get the isolate from script_state->GetIsolate()